internal/lifecycle/lifecycle.go

```diff
@@ -22,18 +22,34 @@ package lifecycle
 
 import (
 	"context"
+	"sort"
 
 	"go.uber.org/fx/internal/fxlog"
 	"go.uber.org/fx/internal/fxreflect"
 	"go.uber.org/multierr"
 )
 
+// Priority of a Hook in the Lifecycle queue
+type Priority int
+
+const (
+	// PriorityLowest hooks are executed last.
+	// For example, hooks that start serving network traffic might wish to execute
+	// only after all other application components are initialized.
+	PriorityLowest Priority = -1000
+	// PriorityNormal is the default priority level
+	PriorityNormal = 0
+	// PriorityHighest hooks are executed first.
+	PriorityHighest = 1000
+)
+
 // A Hook is a pair of start and stop callbacks, either of which can be nil,
 // plus a string identifying the supplier of the hook.
 type Hook struct {
-	OnStart func(context.Context) error
-	OnStop  func(context.Context) error
-	caller  string
+	OnStart  func(context.Context) error
+	OnStop   func(context.Context) error
+	Priority Priority
```

Comment:
This line is technically a breaking change

Reply:
Yes it is a breaking change. If this structure is embedded anonymous with another structure with the field Priority
https://play.golang.org/p/TlpCWwQIjvP

Reply:
Hey! So, embedding of structs is not covered by the [Go 1 Compatibility Expectations](https://golang.org/doc/go1compat) so this wouldn't be a breaking change if we were to make it. Otherwise we would never be able to add new fields and methods to existing structs.